### PR TITLE
[NUI] Async text feature

### PR DESCRIPTION
**Pull request by wonrst**
# Async text feature

## Related patches
* https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-adaptor/+/315160/
* https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-toolkit/+/315140/
* https://review.tizen.org/gerrit/#/c/platform/core/uifw/dali-csharp-binder/+/315818/

## Added RenderMode property
* Sync : default, synchronous text loading.
* AsyncAuto : automatically requests an asynchronous text load in OnRelayout.
* AsyncManual : users should manually request rendering using the async text method.

* All text rendering processes (update/layout/render) are performed asynchronously in AsyncAuto and AsyncManual.

## Added manual async render method
* RequestAsyncRenderWithFixedSize : Requests asynchronous rendering of text with a fixed size.

* RequestAsyncRenderWithFixedWidth : Requests asynchronous text rendering with a fixed width.
The height is determined by the content of the text when rendered with the given width.
If the heightConstraint is given, the maximum height will be the heightConstraint.

* RequestAsyncRenderWithConstraint : Requests asynchronous rendering with the maximum available width using the given widthConstraint.
If the heightConstraint is given, the maximum height will be the heightConstraint.

## Added async size compuation method
* RequestAsyncNaturalSize : Requests asynchronous text natural size computation.

* RequestAsyncHeightForWidth : Requests asynchronous computation of the height of the text based on the given width.

## Added event
* AsyncTextRendered,
* AsyncNaturalSizeComputed,
* AsyncHeightForWidthComputed

## Test & Video
I cannot upload demo video to public github.
* https://github.sec.samsung.net/bowon-ryu/NUI_Text

* https://github.com/wonrst/NUI-Text/tree/main/async_text_1_tpk
* https://github.com/wonrst/NUI-Text/tree/main/async_text_2_tpk

# More information
https://confluence.sec.samsung.net/pages/viewpage.action?spaceKey=GFX&title=Async+Text+Load

### CODE REVIEW

File: src/Tizen.NUI/src/public/BaseComponents/TextLabelEvent.cs [hunk 1]
```diff
@@ -35,12 +35,151 @@ public partial class TextLabel
...
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public event EventHandler<AsyncTextSizeComputedEventArgs> AsyncHeightForWidthComputed
+        {
+            add
+            {
+                if (textLabelAsyncHeightForWidthComputedEventHandler == null)
+                {
+                    textLabelAsyncHeightForWidthComputedCallbackDelegate = (OnAsyncHeightForWidthComputed);
+                    AsyncHeightForWidthComputedSignal().Connect(textLabelAsyncHeightForWidthComputedCallbackDelegate);
```
Comment (hinohie) [inline src/Tizen.NUI/src/public/BaseComponents/TextLabelEvent.cs:74]: I think TextLabelSignal only works with non-argment signal (like TextFit).

When you see `Dali_Signal_Sl_void_Sp_Dali_Toolkit_TextLabel_SP__Sg__Connect` implementation, it only assume that
arg2 is `void (*)(Dali::Toolkit::TextLabel)` type. I think we'd better create new Signal class for it.

File: src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs [hunk 2]
```diff
@@ -217,6 +217,9 @@ static TextLabel()
 
                 CutoutProperty = BindableProperty.Create(nameof(Cutout), typeof(bool), typeof(TextLabel), false,
                     propertyChanged: SetInternalCutoutProperty, defaultValueCreator: GetInternalCutoutProperty);
+
+                RenderModeProperty = BindableProperty.Create(nameof(RenderMode), typeof(TextRenderMode), typeof(TextLabel), TextRenderMode.Sync,
+                    propertyChanged: SetInternalRenderModeProperty, defaultValueCreator: GetInternalRenderModeProperty);
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs:222]: 저희가 더 이상 binderable property를 추가하지 않기로 하였습니다. 
일반 C# property로 구현해주세요

File: src/Tizen.NUI/src/public/BaseComponents/TextLabelEvent.cs (hunk 1 above)
Comment (wonrst) [reply src/Tizen.NUI/src/public/BaseComponents/TextLabelEvent.cs]: 생각지 못한 걸 찾아주셨네요,
수정했습니다.
async 관련 외 signal 들 도 다른 PR에서 정리를 한번 해야겠습니다. 감사합니다.

File: src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs (hunk 2 above)
Comment (wonrst) [reply src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs]: binderable property 제거했습니다. 감사합니다.

**Review by hinohie — APPROVED** [review]

**Review by Seoyeon2Kim — APPROVED** [review]

**Pull request merged**